### PR TITLE
support acs gen2

### PR DESCRIPTION
**Pull request by leonard520**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yanzhudd** [comment]
Please fix the CI

File: src/spring/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 Release History
 ===============
+1.14.0
+---
+* Add new command -- `az spring application-configuration-service create --generation` to support create Application Configuration Service with different generation
+* Add new command -- `az spring application-configuration-service update --generation` to support update Application Configuration Service to different generation
+* Add new command -- `az spring application-configuration-service git repo add --ca-cert-name` to support bind certificate to Application Configuration Service Gen2
```
Comment (yanzhudd) [inline src/spring/HISTORY.md:7]:
```suggestion
* Add new command -- `az spring application-configuration-service create --generation` to support creating Application Configuration Service with different generation
* Add new command -- `az spring application-configuration-service update --generation` to support updating Application Configuration Service to different generation
* Add new command -- `az spring application-configuration-service git repo add --ca-cert-name` to support binding certificate to Application Configuration Service Gen2
```

File: src/spring/azext_spring/_params.py [hunk 2]
```diff
@@ -156,7 +156,14 @@ def load_arguments(self, _):
...
                    options_list=['--enable-application-configuration-service', '--enable-acs'],
+                   arg_group="Application Configuration Service",
                    help='(Enterprise Tier Only) Enable Application Configuration Service.')
+        c.argument('application_configuration_service_generation',
+                   arg_group="Application Configuration Service",
+                   type=ConfigurationServiceGeneration,
+                   options_list=['--application-configuration-service-generation', '--acs-gen'],
+                   validator=validate_acs_create,
+                   help='(Enterprise Tier Only) Application Configuration Service Generation to enable. Allowed values are: ' + ', '.join(list(ConfigurationServiceGeneration)))
         c.argument('enable_application_live_view',
```
Comment (yanzhudd) [inline src/spring/azext_spring/_params.py:167]: If this parameter has limited values, please add `arg_type=get_enum_type(ConfigurationServiceGeneration)` for the parameter and it is not necessay to specify the avaiable values manually in `help`.
```suggestion
        c.argument('application_configuration_service_generation',
                   arg_group="Application Configuration Service",
                   arg_type=get_enum_type(ConfigurationServiceGeneration),
                   options_list=['--application-configuration-service-generation', '--acs-gen'],
                   validator=validate_acs_create,
                   help='(Enterprise Tier Only) Application Configuration Service Generation to enable.)
        c.argument('enable_application_live_view',
```

File: src/spring/azext_spring/_params.py [hunk 3]
```diff
@@ -810,6 +817,10 @@ def prepare_logs_argument(c):
         with self.argument_context('spring application-configuration-service {}'.format(scope)) as c:
             c.argument('app', app_name_type, help='Name of app.', validator=validate_app_name)
 
+    for scope in ['create', 'update']:
+        with self.argument_context('spring application-configuration-service {}'.format(scope)) as c:
+            c.argument('generation', type=ConfigurationServiceGeneration, help='Generation. Allowed values are: ' + ', '.join(list(ConfigurationServiceGeneration)))
```
Comment (yanzhudd) [inline src/spring/azext_spring/_params.py:822]: Same as above

File: src/spring/azext_spring/_validators_enterprise.py [hunk 4]
```diff
@@ -345,6 +345,23 @@ def _is_valid_app_and_profile_name(pattern):
     return len(parts) == 2 and _is_valid_app_name(parts[0]) and _is_valid_profile_name(parts[1])
 
 
+def validate_acs_create(namespace):
+    if namespace.application_configuration_service_generation is not None:
+        if namespace.enable_application_configuration_service is False:
+            raise ArgumentUsageError("--application_configuration_service_generation can only be set when enable application configuration service.")
```
Comment (yanzhudd) [inline src/spring/azext_spring/_validators_enterprise.py:351]:
```suggestion
            raise ArgumentUsageError("--application-configuration-service-generation can only be set when enable application configuration service.")
```

File: src/spring/azext_spring/_validators_enterprise.py [hunk 5]
```diff
@@ -345,6 +345,23 @@ def _is_valid_app_and_profile_name(pattern):
...
+
+def validate_acs_create_or_update(namespace):
+    _validate_acs_generation(namespace)
+
+
+def _validate_acs_generation(namespace):
+    if namespace.generation is None:
+        return
+    if (namespace.generation not in list(ConfigurationServiceGeneration)):
+        raise InvalidArgumentValueError("Allowed generation are: " + ', '.join(list(ConfigurationServiceGeneration)))
```
Comment (yanzhudd) [inline src/spring/azext_spring/_validators_enterprise.py:362]: After adding `arg_type=get_enum_type()` to the parameter, it is not necessary to valid the input values manually.

**Comment by yanzhudd** [comment]
If there is new command added, please add test for the command.

File: src/spring/HISTORY.md (hunk 1 above)
Comment (leonard520) [reply src/spring/HISTORY.md]: updated

File: src/spring/azext_spring/_params.py (hunk 2 above)
Comment (leonard520) [reply src/spring/azext_spring/_params.py]: Thanks! updated

File: src/spring/azext_spring/_params.py (hunk 3 above)
Comment (leonard520) [reply src/spring/azext_spring/_params.py]: updated

File: src/spring/azext_spring/_validators_enterprise.py (hunk 4 above)
Comment (leonard520) [reply src/spring/azext_spring/_validators_enterprise.py]: fixed

File: src/spring/azext_spring/_validators_enterprise.py (hunk 5 above)
Comment (leonard520) [reply src/spring/azext_spring/_validators_enterprise.py]: updated

File: src/spring/azext_spring/_validators.py [hunk 6]
```diff
@@ -106,6 +106,8 @@ def _check_tanzu_components_not_enable(cmd, namespace):
         raise ArgumentUsageError('--enable-application-live-view {}'.format(suffix))
     if namespace.enable_application_accelerator:
         raise ArgumentUsageError('--enable-application-accelerator {}'.format(suffix))
+    if namespace.application_configuration_service_generation:
+        raise ArgumentUsageError('--application_configuration_service_generation {}'.format(suffix))
```
Comment (yanzhudd) [inline src/spring/azext_spring/_validators.py:110]:
```suggestion
        raise ArgumentUsageError('--application-configuration-service-generation {}'.format(suffix))
```

Comment (yanzhudd) [reply src/spring/azext_spring/_validators.py:110]: Besides, it is suggested to point out the error in more details. It is better to specify how to input.

**Comment by yanzhudd** [comment]
Do you want to release a new extension version? If so, please add the version to setup.py.

**Review by yanzhudd — APPROVED** [review]

File: src/spring/HISTORY.md [hunk 7]
```diff
@@ -28,6 +34,7 @@ Release History
 1.12.2
 ---
 * Add default `enabled_state` for `az spring config-server set` in Standard Counsumption tier.
+>>>>>>> main
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:37]:
```suggestion
```

File: src/spring/HISTORY.md [hunk 8]
```diff
@@ -1,5 +1,11 @@
 Release History
 ===============
+1.14.0
+---
+* Add new command -- `az spring application-configuration-service create --generation` to support creating Application Configuration Service with different generation
+* Add new command -- `az spring application-configuration-service update --generation` to support updating Application Configuration Service to different generation
+* Add new command -- `az spring application-configuration-service git repo add --ca-cert-name` to support binding certificate to Application Configuration Service Gen2
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:7]:
```suggestion
* Add new command `az spring application-configuration-service create --generation` to support creating Application Configuration Service with different generation
* Add new command `az spring application-configuration-service update --generation` to support updating Application Configuration Service to different generation
* Add new command `az spring application-configuration-service git repo add --ca-cert-name` to support binding certificate to Application Configuration Service Gen2
```

File: src/spring/azext_spring/_params.py [hunk 9]
```diff
@@ -810,6 +817,10 @@ def prepare_logs_argument(c):
         with self.argument_context('spring application-configuration-service {}'.format(scope)) as c:
             c.argument('app', app_name_type, help='Name of app.', validator=validate_app_name)
 
+    for scope in ['create', 'update']:
+        with self.argument_context('spring application-configuration-service {}'.format(scope)) as c:
+            c.argument('generation', arg_type=get_enum_type(ConfigurationServiceGeneration), help='Generation.')
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_params.py:822]: > help='Generation.'

Could you please provide the more specific and understandable help message?

**Review by yanzhudd — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**